src/data/data.py

```diff
@@ -57,180 +49,12 @@ def load_data_img(args):
-
-        Args:
-            dataframe (pandas.DataFrame): Input dataframe
-            tokenizer (transformers.tokenizer): Transformers tokenizer
-            source_len (int): Max length of source text
-            target_len (int): Max length of target text
-            source_text (str): column name of source text
-            target_text (str): column name of target text
-        """
-        self.tokenizer = tokenizer
-        self.data = {qid: problems[qid] for qid in qids}
-        self.source_len = source_len
-        self.summ_len = target_len
-        self.target_text = []
-        self.source_text = []
-        self.image_ids = []
-        if test_le is not None:
-            test_le_data = json.load(open(test_le))["preds"]
-        else:
-            test_le_data = None
-        idx = 0
-        for qid in self.data:
-            if test_le_data is not None:
-                curr_le_data = test_le_data[idx]
-                idx += 1
-            else:
-                curr_le_data = None
-            prompt, target = build_train_pair(
-                problems, qid, args, curr_le_data)
-            self.target_text.append(target)
-            self.source_text.append(prompt)
-            if str(qid) in name_maps:
-                i_vectors = image_features[int(name_maps[str(qid)])]
-                self.image_ids.append(i_vectors)
-            else:
-                shape = img_shape[args.img_type]
-                self.image_ids.append(np.zeros(shape))
-
-    def __len__(self):
-        """returns the length of dataframe"""
-
-        return len(self.target_text)
-
-    def __getitem__(self, index):
-        """return the input ids, attention masks and target ids"""
-
-        source_text = str(self.source_text[index])
-        target_text = str(self.target_text[index])
-        image_ids = self.image_ids[index]
-
-        # cleaning data so as to ensure data is in string type
-        source_text = " ".join(source_text.split())
-        target_text = " ".join(target_text.split())
-
-        source = self.tokenizer.batch_encode_plus(
-            [source_text],
-            max_length=self.source_len,
-            pad_to_max_length=True,
-            truncation=True,
-            padding="max_length",
-            return_tensors="pt",
-        )
-        target = self.tokenizer.batch_encode_plus(
-            [target_text],
-            max_length=self.summ_len,
-            pad_to_max_length=True,
-            truncation=True,
-            padding="max_length",
-            return_tensors="pt",
-        )
-        source_ids = source["input_ids"].squeeze()
-        source_mask = source["attention_mask"].squeeze()
-        target_ids = target["input_ids"].squeeze().tolist()
-
-        image_ids = torch.tensor(image_ids).squeeze()
-
-        return {
-            "input_ids": source_ids,
-            "attention_mask": source_mask,
-            "image_ids": image_ids,
-            "labels": target_ids,
-        }
```

Comment:
We can probably implement these classes with template methods.
Let's take note and do it in the next iteration